subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/versionupgrade/UpgradeDependencyPlugin.java

```diff
@@ -369,4 +375,15 @@ public UpgradeDependencyExtension getUpgradeDependencyExtension() {
         return upgradeDependencyExtension;
     }
 
+    private void addLazyDependencyVaidation(final String dependency, Task task) {
+        LazyConfiguration.lazyConfiguration(task, new Runnable() {
+            @Override
+            public void run() {
+                if (dependency == null) {
```

Comment:
Since **dependency** will not change its value (it will be null at the moment when addLazyDependencyValidation is called), we can always throw inside lazyConfiguration, but initialize it only when **dependency** is null.